standard/statements.md

```diff
@@ -517,6 +517,12 @@ A *local_function_declaration* may include one `async` ([§15.15](classes.md#151
 
 A local function is declared at block scope, and that function may capture variables from the enclosing scopes. It is a compile-time error if a captured variable is read by the body of the local function but is not definitely assigned before each call to the function. The compiler shall determine which variables are definitely assigned on return ([§9.4.4.33](variables.md#94433-rules-for-variables-in-local-functions)).
 
+When the type of `this` is a struct type, it is a compile-time error for the body of a local function to access `this`. This is true whether the access is explicit (as in `this.x`) or implicit (as in `x` where `x` is an instance member of the struct). This rule only prohibits such access and does not affect whether member lookup results in a member of the struct.
```

Comment:
Do you think we need to explicitly state that assignment to `this` is also prohibited? I think assignment requires access, so we're good, but wanted your thoughts as well.

```csharp
public struct SThis
{
    public void M()
    {
        void LocalFunction()
        {
            this = new SThis();
        }
    }
}
```